### DISCUSSION TITLE
Any method so when a user clicks on data point, it loads other plots and data summaries for that specific point?

### DISCUSSION THREAD

**Discussion by ctrejocaltech** [vega/altair#3113] [Q&A]
I'm working on a histogram that will show the masses of black hole mergers, however I would like it so that if a user clicks on a specific event, it will populate on the same page, specific data and plots for that particular event. 

Here is my code for the histogram: 
```
# Fetch the data from the URL and load it into a DataFrame
url = 'https://gwosc.org/eventapi/csv/GWTC/'
data = pd.read_csv(url)

#try search field
dropdown = alt.binding_select(
    options=['total_mass_source', 'mass_1_source', 'mass_2_source'],
    name='Mass Source '

)

xcol_param = alt.param(
    value='total_mass_source',
    bind=dropdown
)

#tooltip = ['total_mass_source', 'commonName', 'GPS', 'final_mass_source'] #check why this scales the y-axis down
# Create the Altair histogram
hist = alt.Chart(data, title='Histogram of Total Mass').mark_bar().encode(
    #alt.X('total_mass_source:Q', bin=alt.Bin(step=1), title='Total Mass'),
    alt.X('x:Q').title(''),
    alt.Y('count()', scale=alt.Scale(domain=[0, 5]), title='Count'),
    ).transform_calculate(
        x=f'datum[{xcol_param.name}]'
    ).add_params(
        xcol_param
    ).interactive()


# Show the chart
st.altair_chart(hist)
```

so an example of what I have in mind: 
https://catalog.cardiffgravity.org/

but I want to generate more detailed plots for each category, there is already an app that does this: 

https://gw-quickview.streamlit.app

so basically, plot all those events into a histogram, which I did, but then have it output the types of graphs seen in that last link, under the histogram. Is anything like this possible?

**Comment by jonmmease** [comment]
Hi @ctrejocaltech, are you hoping to use a particular dashboarding toolkit? Or are you working in a notebook environment?

I believe this will be possible using [Panel](https://panel.holoviz.org/reference/panes/Vega.html), [Streamlit](https://github.com/domoritz/streamlit-vega-lite), and [Solara](https://solara.dev/api/altair).

**Comment by ctrejocaltech** [comment]
I'm using a notebook environment, I will checkout those links you sent, they do look promising, thank you!

**Reply by jonmmease** [reply]
Today, the panel approach is probably your best bet for use in the notebook. 

We are working on first-class Jupyter Widget support in Altair, so hopefully it will be built-in soon!

**Reply by ctrejocaltech** [reply]
yeah that seems like it might work, thanks!
